funlib/persistence/arrays/ome_datasets.py

```diff
@@ -0,0 +1,199 @@
+import logging
+from pathlib import Path
+from typing import Sequence
+
+from iohub.ngff import TransformationMeta, open_ome_zarr
+from iohub.ngff.models import AxisMeta
+from numpy.typing import DTypeLike
+
+from funlib.geometry import Coordinate
+
+from .array import Array
+from .metadata import MetaData, OME_MetaDataFormat
+from .ome_tmp import get_effective_scale, get_effective_translation
+
+logger = logging.getLogger(__name__)
+
+
+def open_ome_ds(
+    store: Path,
+    name: str,
+    mode: str = "r",
+    **kwargs,
+) -> Array:
+    """
+    Open an ome-zarr dataset with common metadata that is useful for indexing with physcial coordinates.
+
+    Args:
+
+        store:
+
+            See https://czbiohub-sf.github.io/iohub/main/api/ngff.html#iohub.open_ome_zarr
+
+        name:
+
+            The name of the dataset in your ome-zarr dataset.
+
+        mode:
+
+            See https://zarr.readthedocs.io/en/stable/api/convenience.html#zarr.convenience.open
+
+        kwargs:
+
+            See additional arguments available here:
+            https://czbiohub-sf.github.io/iohub/main/api/ngff.html#iohub.open_ome_zarr
+
+
+    Returns:
+
+        A :class:`Array` supporting spatial indexing on your dataset.
+    """
+
+    assert (store / name).exists(), "Store does not exist!"
```

Comment:
This is personal preference but putting the store path into the failure message is always super helpful

Reply:
Done